extensions/hibernate-envers/deployment/src/main/java/io/quarkus/hibernate/envers/deployment/HibernateEnversProcessor.java

```diff
@@ -34,18 +35,20 @@ List<AdditionalJpaModelBuildItem> addJpaModelClasses() {
     @BuildStep
     public void registerEnversReflections(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             HibernateEnversBuildTimeConfig buildTimeConfig) {
-        reflectiveClass.produce(ReflectiveClassBuildItem.builder("org.hibernate.envers.DefaultRevisionEntity").methods()
-                .build());
-        reflectiveClass
-                .produce(ReflectiveClassBuildItem.builder("org.hibernate.envers.DefaultTrackingModifiedEntitiesRevisionEntity")
-                        .methods().build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(
+                "org.hibernate.envers.DefaultRevisionEntity",
+                "org.hibernate.envers.DefaultTrackingModifiedEntitiesRevisionEntity")
+                .reason(getClass().getName())
```

Comment:
This seems very repetitive... Can't we imagine Quarkus would allow injecting the source build step (class/method name) into a build item somehow, and use that as the default reason?

I'm pretty sure we can, but @dmlloyd may be needed to explain us whether it would be a good idea or not :)

Reply:
Yup, that's quite repetitive and boring (thankfully I could use the IDE to do it for the most part) :), but I would expect us to add something more descriptive in most cases, e.g. instead of just showing what class added it I would expect people to also add a short comment on why it's needed.